Proposal: Kyverno to Graduation (#1217)
cncf/toc: .github/ISSUE_TEMPLATE/graduation.md

Completion of this due diligence document, resolution of concerns raised, and presented for public comment satisifies the Due Diligence Review criteria.

- [x] **Additional documentation as appropriate for project type, e.g.: installation documentation, end user documentation, reference implementation and/or code samples.**

<!-- (Project assertion goes here) --> 

## Governance and Maintainers

Note: this section may be augmented by the completion of a Governance Review from TAG Contributor Strategy.

### Suggested

- [x]  **Governance has continuously been iterated upon by the project as a result of their experience applying it, with the governance history demonstrating evolution of maturity alongside the project's maturity evolution.**

<!-- (Project assertion goes here) --> 

The strategies for major Project Governance were discussed and refined during the Kyverno maintainers' meeting. You can find the meeting notes [here](https://docs.google.com/document/d/1I_GWsz32gLw8sQyuu_Wv0-WQrtRLjn9FuX2KGNkvUY4/edit):
* 05-21-2024
* 02-20-2024
* 12-05-2023

The Governance page has been maintained and updated through Kyverno website's Github repo, history:

> https://github.com/kyverno/website/commits/main/content/en/community/_index.md

Reviewer: ```suggestion
https://kyverno.io/community/
```

Author: Pointed to the `kyverno/community` repo.

### Required

> - [ ] **Clear and discoverable project governance documentation.**

Reviewer: ```suggestion
- [x] **Clear and discoverable project governance documentation.**
```

Author: Nice catch, updated!

<!-- (Project assertion goes here) --> 

Governance docs are up to date and available at https://kyverno.io/community/.

- [ ] **Governance clearly documents [vendor-neutrality](https://contribute.cncf.io/maintainers/community/vendor-neutrality/) of project direction.**

<!-- (Project assertion goes here) --> 

- [ ] **Document how the project makes decisions on leadership roles, contribution acceptance, requests to the CNCF, and changes to governance or project goals.**

<!-- (Project assertion goes here) --> 

- [x] **Document how role, function-based members, or sub-teams are assigned, onboarded, and removed for specific teams (example: Security Response Committee).**

<!-- (Project assertion goes here) --> 

[Project roles](https://kyverno.io/community/#project-roles).

- [x] **Document complete list of current maintainers, including names, contact information, domain of responsibility, and affiliation.**

<!-- (Project assertion goes here) --> 

[Maintainers](https://github.com/kyverno/kyverno/blob/main/MAINTAINERS.md).

- [x] **A number of active maintainers which is appropriate to the size and scope of the project.**

<!-- (Project assertion goes here) --> 

> The Kyverno project currently has eight (8) maintainers from four (4) affiliations.

Reviewer: ```suggestion
The Kyverno project currently has five (5) maintainers from one (1) affiliation.
```

- [x] **Document a complete maintainer lifecycle process (including roles, onboarding, offboarding, and emeritus status).**

<!-- (Project assertion goes here) --> 

This is documented under [Maintainers](https://kyverno.io/community/#maintainers) section.

- [x] **Demonstrate usage of the maintainer lifecycle with outcomes, either through the addition or replacement of maintainers as project events have required.**

<!-- (Project assertion goes here) --> 

The process of onboarding a maintainer is initiated by submitting a pull request. 

For example, [feat: add myself (vishal-chdhry) to maintainers list #9125](https://github.com/kyverno/kyverno/pull/9125).

The same with offboarding process, and emeritus maintainers are listed at: 

https://github.com/kyverno/kyverno/blob/main/MAINTAINERS.md#maintainers-emeritus

> - [x] **Project maintainers from at least 2 organizations that demonstrates survivability.**

Reviewer: ```suggestion
- [ ] **Project maintainers from at least 2 organizations that demonstrates survivability.**
```

<!-- (Project assertion goes here) --> 

> Currently maintainers are from four (4) organizations including Nirmata, Stackwatch, Ohio Supercomputer Center, Giant Swarm GmbH.

Reviewer: ```suggestion
Currently maintainers are from one (1) organization, Nirmata.
```

    - [x] [CODEOWNERS](https://github.com/kyverno/kyverno/blob/main/CODEOWNERS)
    - [ ] doc owners?

<!-- (Project assertion goes here) --> 

- [x] **Document agreement that project will adopt CNCF Code of Conduct.**

<!-- (Project assertion goes here) --> 

Kyverno follows the [Code of Conduct](https://github.com/kyverno/kyverno/blob/main/CODE_OF_CONDUCT.md), which is aligned with the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).

- [x] **CNCF Code of Conduct is cross-linked from other governance documents.**

<!-- (Project assertion goes here) -->

This is documented on Github [Code of Conduct](https://github.com/kyverno/kyverno/blob/main/GOVERNANCE.md#code-of-conduct) and Kyverno website under Project Governance, see [here](https://kyverno.io/community/#project-governance).

- [ ] **All subprojects, if any, are listed.**

<!-- (Project assertion goes here) --> 

- [x] **If the project has subprojects: subproject leadership, contribution, maturity status documented, including add/remove process.**

<!-- (Project assertion goes here) --> 

> This is clarified in the [Project Governance section](https://main.kyverno.io/community/#project-governance).

Reviewer: ```suggestion
This is clarified in the [Project Governance section](https://kyverno.io/community/#project-governance).
```

Author: The project governance has been moved to `kyverno/community` GitHub repo, updated to https://github.com/kyverno/community/blob/main/GOVERNANCE.md.
